### PR TITLE
Implement it

### PR DESCRIPTION
**Pull request by trieloff**
- feat: first implementation
- test(post-deploy): add post-deploy test
- build(test): exit mocha
- build(test): do not require full branch coverage
- test: format fixture

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -9,24 +9,36 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-const { wrap } = require('@adobe/helix-shared');
+const wrap = require('@adobe/helix-shared-wrap');
+const { optionalConfig } = require('@adobe/helix-shared-config');
```
Comment (trieloff) [inline src/index.js:13]: @tripodsan I must say I was skeptical regarding the repo split, but it's nice to see it in action.

File: src/index.js [hunk 2]
```diff
@@ -9,24 +9,36 @@
...
-  const name = new URL(request.url).searchParams.get('name') || 'world';
-  context.log.info(`Saying hello to: ${name}.`);
-  return new Response(`Hello, ${name}.`);
+  const aggregate = Object.keys(context.config).reduce((agg, name) => {
+    if (context.config[name] && typeof context.config[name].toJSON === 'function') {
+      // eslint-disable-next-line no-param-reassign
+      agg[name] = context.config[name].toJSON();
+    }
+    return agg;
+  }, {});
```
Comment (dominique-pfister) [inline src/index.js:31]: Ah, `reduce` in action 😉

**Review by dominique-pfister — APPROVED** [review]

File: src/index.js (hunk 2 above)
Comment (trieloff) [reply src/index.js:31]: Yep, not a single `let` needed.

File: src/index.js [hunk 3]
```diff
@@ -9,24 +9,36 @@
...
+    },
+  });
 }
 
 module.exports.main = wrap(main)
   .with(status)
   .with(logger.trace)
-  .with(logger);
+  .with(logger)
+  .with(optionalConfig, 'redirect', 'fstab', 'markup', 'index');
```
Comment (tripodsan) [inline src/index.js]: do we really need `markup` ?

**Review by tripodsan — COMMENTED** [review]
LGTM (except the reduce :-) ). 

but can we also include the 
- `head.html` ?
- `helix-version.txt` ?

@davidnuescheler ideally, we could also consider the existence of a metadata.json as config. i.e. just add a:

```
"metadata": "/metadata.json`
```

entry. so we don't need to fetch a 404 if there is none.

File: src/index.js [hunk 4]
```diff
@@ -10,23 +10,35 @@
...
+  return new Response(JSON.stringify(aggregate), {
+    headers: {
+      'content-type': 'application/json',
+    },
+  });
 }
 
 module.exports.main = wrap(main)
   .with(status)
+  .with(optionalConfig, 'redirect', 'fstab', 'markup', 'index')
```
Comment (tripodsan) [inline src/index.js:42]: @trieloff I moved this up, so that logging is enabled

**Pull request merged**